boards/soc: Rename to use vendor prefixes and misc. fixes

nordicjm opened the pull request:
Some changes that were missed in the collab-hwm branch.

Compliance failure is expected due to large images being moved into different folder names

nordicjm committed:
boards/socs: Rename folders to have proper vendor prefix in

Replaces inaccurate or wrong vendor prefixes in board and soc
folder names with those from thr vendor prefix file

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
soc: Move non-grouped qemu boards into qemu folder

For the 2 SoCs without a vendor, put them into a generic qemu
folder

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
dts: bindings: vendor-prefixes: Add additional vendor prefixes

Adds additional vendor prefixes which have been used for boards

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
maintainers: Update paths

Updates paths for renamed folders

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
boards: Fix invalid paths

Fixes paths that are not valid

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

tejlmand reviewed (commented):
Only concern is that `up` as vendor prefix seems to be a little too generic.

nordicjm commented on dts/bindings/vendor-prefixes.txt:
```diff
@@ -678,6 +682,7 @@ udoo	Udoo
 ugoos	Ugoos Industrial Co., Ltd.
 ultrachip	UltraChip Inc.
 uniwest	United Western Technologies Corp (UniWest)
+up	UP Bridge the Gap
```
Changed to `up-bridge-the-gap`

nordicjm committed:
boards: Remove setting CONFIG_BOARD

Removes instances of setting CONFIG_BOARD that should have been
removed in the hwmv2 porting process

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

henrikbrixandersen commented on dts/bindings/vendor-prefixes.txt:
```diff
@@ -72,6 +73,7 @@ arm	ARM Ltd.
 armadeus	ARMadeus Systems SARL
 arrow	Arrow Electronics
 artesyn	Artesyn Embedded Technologies Inc.
+arturo182	Arturo182
```
This vendor should like be "solderparty" or similar, arturo182 is just his username. Another option would be to put them under "others".

https://www.tindie.com/stores/arturo182/

nordicjm replied:
Fixed

henrikbrixandersen reviewed (dismissed):
Awesome. One minor suggestion.

carlescufi commented:
Compliance failure is about existing image size.

henrikbrixandersen commented:
@nordicjm Merge conflict in `boards/synopsys/nsim/board.cmake`.
